from PyQt5 import QtCore, QtGui, QtWidgets
from Variables import Variables

class Ui_dialog_Setting(object):
    def setupUi(self, dialog_Setting):
        dialog_Setting.setObjectName("dialog_Setting")
        dialog_Setting.setFixedSize(350, 370)

        palette = QtGui.QPalette()
        brush = QtGui.QBrush(QtGui.QColor(0, 0, 0))
        brush.setStyle(QtCore.Qt.SolidPattern)
        palette.setBrush(QtGui.QPalette.Active, QtGui.QPalette.WindowText, brush)
        brush = QtGui.QBrush(QtGui.QColor(85, 170, 255))
        brush.setStyle(QtCore.Qt.SolidPattern)
        palette.setBrush(QtGui.QPalette.Active, QtGui.QPalette.Button, brush)
        brush = QtGui.QBrush(QtGui.QColor(85, 170, 255))
        brush.setStyle(QtCore.Qt.SolidPattern)
        palette.setBrush(QtGui.QPalette.Active, QtGui.QPalette.Window, brush)
        brush = QtGui.QBrush(QtGui.QColor(0, 0, 0))
        brush.setStyle(QtCore.Qt.SolidPattern)
        palette.setBrush(QtGui.QPalette.Inactive, QtGui.QPalette.WindowText, brush)
        brush = QtGui.QBrush(QtGui.QColor(85, 170, 255))
        brush.setStyle(QtCore.Qt.SolidPattern)
        palette.setBrush(QtGui.QPalette.Inactive, QtGui.QPalette.Button, brush)
        brush = QtGui.QBrush(QtGui.QColor(85, 170, 255))
        brush.setStyle(QtCore.Qt.SolidPattern)
        palette.setBrush(QtGui.QPalette.Inactive, QtGui.QPalette.Window, brush)
        brush = QtGui.QBrush(QtGui.QColor(42, 85, 127))
        brush.setStyle(QtCore.Qt.SolidPattern)
        palette.setBrush(QtGui.QPalette.Disabled, QtGui.QPalette.WindowText, brush)
        brush = QtGui.QBrush(QtGui.QColor(85, 170, 255))
        brush.setStyle(QtCore.Qt.SolidPattern)
        palette.setBrush(QtGui.QPalette.Disabled, QtGui.QPalette.Button, brush)
        brush = QtGui.QBrush(QtGui.QColor(85, 170, 255))
        brush.setStyle(QtCore.Qt.SolidPattern)
        palette.setBrush(QtGui.QPalette.Disabled, QtGui.QPalette.Window, brush)
        dialog_Setting.setPalette(palette)
        dialog_Setting.setAutoFillBackground(True)
        self.label_Setting = QtWidgets.QLabel(dialog_Setting)
        self.label_Setting.setGeometry(QtCore.QRect(20, 10, 311, 40))
        font = QtGui.QFont()
        font.setPointSize(16)
        self.label_Setting.setFont(font)
        self.label_Setting.setAlignment(QtCore.Qt.AlignHCenter|QtCore.Qt.AlignTop)
        self.label_Setting.setObjectName("label_Setting")
        self.pushButton_Apply = QtWidgets.QPushButton(dialog_Setting)
        self.pushButton_Apply.setGeometry(QtCore.QRect(240, 320, 101, 31))
        palette = QtGui.QPalette()
        brush = QtGui.QBrush(QtGui.QColor(0, 0, 0))
        brush.setStyle(QtCore.Qt.SolidPattern)
        palette.setBrush(QtGui.QPalette.Active, QtGui.QPalette.WindowText, brush)
        brush = QtGui.QBrush(QtGui.QColor(50, 170, 50))
        brush.setStyle(QtCore.Qt.SolidPattern)
        palette.setBrush(QtGui.QPalette.Active, QtGui.QPalette.Button, brush)
        brush = QtGui.QBrush(QtGui.QColor(50, 170, 50))
        brush.setStyle(QtCore.Qt.SolidPattern)
        palette.setBrush(QtGui.QPalette.Active, QtGui.QPalette.Window, brush)
        brush = QtGui.QBrush(QtGui.QColor(0, 0, 0))
        brush.setStyle(QtCore.Qt.SolidPattern)
        palette.setBrush(QtGui.QPalette.Inactive, QtGui.QPalette.WindowText, brush)
        brush = QtGui.QBrush(QtGui.QColor(50, 170, 50))
        brush.setStyle(QtCore.Qt.SolidPattern)
        palette.setBrush(QtGui.QPalette.Inactive, QtGui.QPalette.Button, brush)
        brush = QtGui.QBrush(QtGui.QColor(50, 170, 50))
        brush.setStyle(QtCore.Qt.SolidPattern)
        palette.setBrush(QtGui.QPalette.Inactive, QtGui.QPalette.Window, brush)
        brush = QtGui.QBrush(QtGui.QColor(25, 85, 25))
        brush.setStyle(QtCore.Qt.SolidPattern)
        palette.setBrush(QtGui.QPalette.Disabled, QtGui.QPalette.WindowText, brush)
        brush = QtGui.QBrush(QtGui.QColor(50, 170, 50))
        brush.setStyle(QtCore.Qt.SolidPattern)
        palette.setBrush(QtGui.QPalette.Disabled, QtGui.QPalette.Button, brush)
        brush = QtGui.QBrush(QtGui.QColor(50, 170, 50))
        brush.setStyle(QtCore.Qt.SolidPattern)
        palette.setBrush(QtGui.QPalette.Disabled, QtGui.QPalette.Window, brush)
        self.pushButton_Apply.setPalette(palette)
        font = QtGui.QFont()
        font.setPointSize(10)
        self.pushButton_Apply.setFont(font)
        self.pushButton_Apply.setAutoFillBackground(True)
        self.pushButton_Apply.setObjectName("pushButton_Apply")
        self.pushButton_Ok = QtWidgets.QPushButton(dialog_Setting)
        self.pushButton_Ok.setGeometry(QtCore.QRect(20, 320, 101, 31))
        palette = QtGui.QPalette()
        brush = QtGui.QBrush(QtGui.QColor(0, 0, 0))
        brush.setStyle(QtCore.Qt.SolidPattern)
        palette.setBrush(QtGui.QPalette.Active, QtGui.QPalette.WindowText, brush)
        brush = QtGui.QBrush(QtGui.QColor(85, 210, 255))
        brush.setStyle(QtCore.Qt.SolidPattern)
        palette.setBrush(QtGui.QPalette.Active, QtGui.QPalette.Button, brush)
        brush = QtGui.QBrush(QtGui.QColor(85, 210, 255))
        brush.setStyle(QtCore.Qt.SolidPattern)
        palette.setBrush(QtGui.QPalette.Active, QtGui.QPalette.Window, brush)
        brush = QtGui.QBrush(QtGui.QColor(0, 0, 0))
        brush.setStyle(QtCore.Qt.SolidPattern)
        palette.setBrush(QtGui.QPalette.Inactive, QtGui.QPalette.WindowText, brush)
        brush = QtGui.QBrush(QtGui.QColor(85, 210, 255))
        brush.setStyle(QtCore.Qt.SolidPattern)
        palette.setBrush(QtGui.QPalette.Inactive, QtGui.QPalette.Button, brush)
        brush = QtGui.QBrush(QtGui.QColor(85, 210, 255))
        brush.setStyle(QtCore.Qt.SolidPattern)
        palette.setBrush(QtGui.QPalette.Inactive, QtGui.QPalette.Window, brush)
        brush = QtGui.QBrush(QtGui.QColor(42, 105, 127))
        brush.setStyle(QtCore.Qt.SolidPattern)
        palette.setBrush(QtGui.QPalette.Disabled, QtGui.QPalette.WindowText, brush)
        brush = QtGui.QBrush(QtGui.QColor(85, 210, 255))
        brush.setStyle(QtCore.Qt.SolidPattern)
        palette.setBrush(QtGui.QPalette.Disabled, QtGui.QPalette.Button, brush)
        brush = QtGui.QBrush(QtGui.QColor(85, 210, 255))
        brush.setStyle(QtCore.Qt.SolidPattern)
        palette.setBrush(QtGui.QPalette.Disabled, QtGui.QPalette.Window, brush)
        self.pushButton_Ok.setPalette(palette)
        font = QtGui.QFont()
        font.setPointSize(10)
        self.pushButton_Ok.setFont(font)
        self.pushButton_Ok.setAutoFillBackground(True)
        self.pushButton_Ok.setObjectName("pushButton_Ok")
        self.pushButton_Cancel = QtWidgets.QPushButton(dialog_Setting)
        self.pushButton_Cancel.setGeometry(QtCore.QRect(130, 320, 101, 31))
        palette = QtGui.QPalette()
        brush = QtGui.QBrush(QtGui.QColor(0, 0, 0))
        brush.setStyle(QtCore.Qt.SolidPattern)
        palette.setBrush(QtGui.QPalette.Active, QtGui.QPalette.WindowText, brush)
        brush = QtGui.QBrush(QtGui.QColor(255, 70, 70))
        brush.setStyle(QtCore.Qt.SolidPattern)
        palette.setBrush(QtGui.QPalette.Active, QtGui.QPalette.Button, brush)
        brush = QtGui.QBrush(QtGui.QColor(255, 70, 70))
        brush.setStyle(QtCore.Qt.SolidPattern)
        palette.setBrush(QtGui.QPalette.Active, QtGui.QPalette.Window, brush)
        brush = QtGui.QBrush(QtGui.QColor(0, 0, 0))
        brush.setStyle(QtCore.Qt.SolidPattern)
        palette.setBrush(QtGui.QPalette.Inactive, QtGui.QPalette.WindowText, brush)
        brush = QtGui.QBrush(QtGui.QColor(255, 70, 70))
        brush.setStyle(QtCore.Qt.SolidPattern)
        palette.setBrush(QtGui.QPalette.Inactive, QtGui.QPalette.Button, brush)
        brush = QtGui.QBrush(QtGui.QColor(255, 70, 70))
        brush.setStyle(QtCore.Qt.SolidPattern)
        palette.setBrush(QtGui.QPalette.Inactive, QtGui.QPalette.Window, brush)
        brush = QtGui.QBrush(QtGui.QColor(127, 35, 35))
        brush.setStyle(QtCore.Qt.SolidPattern)
        palette.setBrush(QtGui.QPalette.Disabled, QtGui.QPalette.WindowText, brush)
        brush = QtGui.QBrush(QtGui.QColor(255, 70, 70))
        brush.setStyle(QtCore.Qt.SolidPattern)
        palette.setBrush(QtGui.QPalette.Disabled, QtGui.QPalette.Button, brush)
        brush = QtGui.QBrush(QtGui.QColor(255, 70, 70))
        brush.setStyle(QtCore.Qt.SolidPattern)
        palette.setBrush(QtGui.QPalette.Disabled, QtGui.QPalette.Window, brush)
        self.pushButton_Cancel.setPalette(palette)
        font = QtGui.QFont()
        font.setPointSize(10)
        self.pushButton_Cancel.setFont(font)
        self.pushButton_Cancel.setAutoFillBackground(True)
        self.pushButton_Cancel.setObjectName("pushButton_Cancel")
        self.groupBox_Fonts = QtWidgets.QGroupBox(dialog_Setting)
        self.groupBox_Fonts.setGeometry(QtCore.QRect(30, 50, 301, 251))
        font = QtGui.QFont()
        font.setFamily("MS Reference Sans Serif")
        font.setPointSize(12)
        self.groupBox_Fonts.setFont(font)
        self.groupBox_Fonts.setObjectName("groupBox_Fonts")
        self.groupBox_ButtonFont = QtWidgets.QGroupBox(self.groupBox_Fonts)
        self.groupBox_ButtonFont.setGeometry(QtCore.QRect(10, 40, 271, 61))
        font = QtGui.QFont()
        font.setFamily("MS Reference Sans Serif")
        font.setPointSize(10)
        self.groupBox_ButtonFont.setFont(font)
        self.groupBox_ButtonFont.setObjectName("groupBox_ButtonFont")
        self.fontComboBox_ButtonFont = QtWidgets.QFontComboBox(self.groupBox_ButtonFont)
        self.fontComboBox_ButtonFont.setGeometry(QtCore.QRect(30, 30, 147, 22))
        self.fontComboBox_ButtonFont.setWritingSystem(1)
        font = QtGui.QFont()
        font.setFamily("MS Reference Sans Serif")
        font.setPointSize(8)
        self.fontComboBox_ButtonFont.setFont(font)
        self.fontComboBox_ButtonFont.setObjectName("fontComboBox_ButtonFont")
        self.spinBox_ButtonFont = QtWidgets.QSpinBox(self.groupBox_ButtonFont)
        self.spinBox_ButtonFont.setGeometry(QtCore.QRect(200, 30, 41, 22))
        font = QtGui.QFont()
        font.setFamily("MS Reference Sans Serif")
        font.setPointSize(8)
        self.spinBox_ButtonFont.setFont(font)
        self.spinBox_ButtonFont.setObjectName("spinBox_ButtonFont")
        self.groupBox_TopLabel = QtWidgets.QGroupBox(self.groupBox_Fonts)
        self.groupBox_TopLabel.setGeometry(QtCore.QRect(10, 110, 271, 61))
        font = QtGui.QFont()
        font.setFamily("MS Reference Sans Serif")
        font.setPointSize(10)
        self.groupBox_TopLabel.setFont(font)
        self.groupBox_TopLabel.setObjectName("groupBox_TopLabel")
        self.fontComboBox_TopLabelFont = QtWidgets.QFontComboBox(self.groupBox_TopLabel)
        self.fontComboBox_TopLabelFont.setGeometry(QtCore.QRect(30, 30, 147, 22))
        self.fontComboBox_TopLabelFont.setWritingSystem(1)
        font = QtGui.QFont()
        font.setFamily("MS Reference Sans Serif")
        font.setPointSize(8)
        self.fontComboBox_TopLabelFont.setFont(font)
        self.fontComboBox_TopLabelFont.setObjectName("comboBox_TopLabelFont")
        self.spinBox_TopLabelFont = QtWidgets.QSpinBox(self.groupBox_TopLabel)
        self.spinBox_TopLabelFont.setGeometry(QtCore.QRect(200, 30, 41, 22))
        font = QtGui.QFont()
        font.setFamily("MS Reference Sans Serif")
        font.setPointSize(8)
        self.spinBox_TopLabelFont.setFont(font)
        self.spinBox_TopLabelFont.setObjectName("spinBox_TopLabelFont")
        self.groupBox_OtherLabel = QtWidgets.QGroupBox(self.groupBox_Fonts)
        self.groupBox_OtherLabel.setGeometry(QtCore.QRect(10, 180, 271, 61))
        font = QtGui.QFont()
        font.setFamily("MS Reference Sans Serif")
        font.setPointSize(10)
        self.groupBox_OtherLabel.setFont(font)
        self.groupBox_OtherLabel.setObjectName("groupBox_OtherLabel")
        self.fontComboBox_OtherLabel = QtWidgets.QFontComboBox(self.groupBox_OtherLabel)
        self.fontComboBox_OtherLabel.setGeometry(QtCore.QRect(30, 30, 147, 22))
        self.fontComboBox_OtherLabel.setWritingSystem(1)
        font = QtGui.QFont()
        font.setFamily("MS Reference Sans Serif")
        font.setPointSize(8)
        self.fontComboBox_OtherLabel.setFont(font)
        self.fontComboBox_OtherLabel.setObjectName("fontComboBox_OtherLabel")
        self.spinBox_OtherLabelFont = QtWidgets.QSpinBox(self.groupBox_OtherLabel)
        self.spinBox_OtherLabelFont.setGeometry(QtCore.QRect(200, 30, 41, 22))
        font = QtGui.QFont()
        font.setFamily("MS Reference Sans Serif")
        font.setPointSize(8)
        self.spinBox_OtherLabelFont.setFont(font)
        self.spinBox_OtherLabelFont.setObjectName("spinBox_OtherLabelFont")

        self.pushButton_Apply.clicked.connect(self.clicked_Apply)
        self.pushButton_Cancel.clicked.connect(self.clicked_Cancel)
        self.pushButton_Ok.clicked.connect(lambda: self.clicked_Ok(dialog_Setting))

        self.LoadFonts()

        

        self.retranslateUi(dialog_Setting)
        QtCore.QMetaObject.connectSlotsByName(dialog_Setting)

    def retranslateUi(self, dialog_Setting):
        _translate = QtCore.QCoreApplication.translate
        dialog_Setting.setWindowTitle(_translate("dialog_Setting", "Select picture"))
        self.label_Setting.setText(_translate("dialog_Setting", "Setting"))
        self.pushButton_Apply.setText(_translate("dialog_Setting", "Apply"))
        self.pushButton_Ok.setText(_translate("dialog_Setting", "Ok"))
        self.pushButton_Cancel.setText(_translate("dialog_Setting", "Cancel"))
        self.groupBox_Fonts.setTitle(_translate("dialog_Setting", "Fonts"))
        self.groupBox_ButtonFont.setTitle(_translate("dialog_Setting", "Button Font"))
        self.groupBox_TopLabel.setTitle(_translate("dialog_Setting", "Top Label Font"))
        self.groupBox_OtherLabel.setTitle(_translate("dialog_Setting", "Other Label Font"))
    
    def LoadFonts(self):
        self.Variable = Variables()
        self.fontComboBox_ButtonFont.setFont(self.Variable.LFont)
        self.fontComboBox_OtherLabel.setFont(self.Variable.LFont)
        self.fontComboBox_TopLabelFont.setFont(self.Variable.LFont)
        self.spinBox_ButtonFont.setFont(self.Variable.LFont)
        self.spinBox_OtherLabelFont.setFont(self.Variable.LFont)
        self.spinBox_TopLabelFont.setFont(self.Variable.LFont)
        self.label_Setting.setFont(self.Variable.TLFont)
        self.pushButton_Apply.setFont(self.Variable.BFont)
        self.pushButton_Cancel.setFont(self.Variable.BFont)
        self.pushButton_Ok.setFont(self.Variable.BFont)
        self.groupBox_Fonts.setFont(self.Variable.LFont)
        self.groupBox_ButtonFont.setFont(self.Variable.LFont)
        self.groupBox_OtherLabel.setFont(self.Variable.LFont)
        self.groupBox_TopLabel.setFont(self.Variable.LFont)

        self.fontComboBox_ButtonFont.setCurrentText(self.Variable.BFamily)
        self.fontComboBox_OtherLabel.setCurrentText(self.Variable.LFamily)
        self.fontComboBox_TopLabelFont.setCurrentText(self.Variable.TLFamily)
        self.spinBox_ButtonFont.setValue(self.Variable.BSize)
        self.spinBox_OtherLabelFont.setValue(self.Variable.LSize)
        self.spinBox_TopLabelFont.setValue(self.Variable.TLSize)

    def clicked_Apply(self):
        TLFamily = self.fontComboBox_TopLabelFont.currentText()
        TLSize = self.spinBox_TopLabelFont.value()
        BFamily = self.fontComboBox_ButtonFont.currentText()
        BSize = self.spinBox_ButtonFont.value()
        LFamily = self.fontComboBox_OtherLabel.currentText()
        LSize = self.spinBox_OtherLabelFont.value()

        self.Variable.ResetFonts(TLFamily, TLSize, BFamily, BSize, LFamily, LSize)
        self.LoadFonts()

    def clicked_Cancel(self):
        pass

    def clicked_Ok(self, dialog_Setting):
        self.clicked_Apply()
        dialog_Setting.close()

    



if __name__ == "__main__":
    import sys
    app = QtWidgets.QApplication(sys.argv)
    app.setStyle('Fusion')
    dialog_Setting = QtWidgets.QDialog()
    ui = Ui_dialog_Setting()
    ui.setupUi(dialog_Setting)
    dialog_Setting.show()
    sys.exit(app.exec_())
